testtools/iothub_test/src/iothub_account.c

```diff
@@ -976,6 +976,19 @@ IOTHUB_PROVISIONED_DEVICE* IoTHubAccount_GetX509Device(IOTHUB_ACCOUNT_INFO_HANDL
     }
 }
 
+IOTHUB_PROVISIONED_DEVICE* IoTHubAccount_GetDevice(IOTHUB_ACCOUNT_INFO_HANDLE acctHandle, IOTHUB_ACCOUNT_AUTH_METHOD accountAuthMethod)
+{
+    if (accountAuthMethod == IOTHUB_ACCOUNT_AUTH_X509)
+    {
+        return IoTHubAccount_GetX509Device(acctHandle);
+    }
+    else
```

Comment:
```suggestion
    else if (accountAuthMethod  == IOTHUB_ACCOUNT_AUTH_CONNSTRING)
```

Reply:
Is this valid for SAS?

Reply:
Fixing the else if for explicit check.

RE question, the connection strings that we use gets generated thanks to the helper `createSASConnectionString`.  However I think this is a misnomer because this connection string includes the `;SharedAccessKey=` and not `SharedAccessSignature`.  Leaving as-is for now.